### DISCUSSION TITLE
Suggestions to get stations with good latest observation data

### DISCUSSION THREAD

**Discussion by 14Slyfox** [weather-gov/api#778] [Q&A]
Great API, and I'm getting everything to work fine.  However, I've come across stations that return no latest observation data.  This is what I'm attempting to do:  Our company owns assets in different states.  I get all the weather stations for those states and narrow the stations needed down (by finding the closest station to our assets based on lat/lon).  Many assets share the same stations so I don't have to pull the same station multiple times.  However, many of these stations do not have "latest observation" data available.  Is there a way to determine what stations are actively updating their observations when I pull the stations?  I would like to only retrieve stations that have current observation data.  It would be nice to have a Boolean flag when getting the station data that the station is current.  If anyone can recommend a solution/workaround it would be much appreciated.

**Comment by dwhitemv25** [comment]
The problem is, "what is the definition of current?"  There's no parameter in API that I know of to specify what "current" is, and the observation time and frequency for each station is different -- some send observations every 15 minutes, some once per hour, and some are variable. 

Can you be more specific as to what "current" is?

**Reply by 14Slyfox** [reply]
When I say latest, I'm not really concerned with how long it was since the last update.  I just want to know that the station that I get from the [/stations] request will return stations that work with the [/stations/{stationId}/observations/latest].  Is there something I should look for from the [/stations] response that will help identify reliable stations?  For example, one of the stations I got back was: AN746.  When I attempt to get the latest observation, I get this response:
Response Body
{
  "correlationId": "17fb94a3",
  "title": "Not Found",
  "type": "https://api.weather.gov/problems/NotFound",
  "status": 404,
  "detail": "Not Found",
  "instance": "https://api.weather.gov/requests/17fb94a3"
}
Response Headers
 cache-control: private, must-revalidate, max-age=271 
 content-length: 225 
 content-type: application/problem+json 
 expires: Wed, 20 Nov 2024 00:02:45 GMT 
 pragma: no-cache 
 x-correlation-id: 18006c50 
 x-request-id: 6255531b-b538-4908-8ebb-693c25cb6fba 
 x-server-id: vm-bldr-nids-apiapp13.ncep.noaa.gov

**Reply by dwhitemv25** [reply]
I see what you mean. Hm, that particular station is non-traditional -- it is registered to the Oklahoma Department of Environmental Quality -- and its observations are missing from API, though I can see them on MesoWest. Out of curiosity, what is the API query that returns this station?

**Reply by 14Slyfox** [reply]
One thing I have noticed is the 4 letter stations seem to be pretty reliable.  The old process we had used the XML files, and I believe the index file that had all of the stations only contained 4 letter stations.  I hate to limit down to just those stations if there are other reliable stations with longer station IDs.

**Comment by 14Slyfox** [comment]
That was returned from this request:
https://api.weather.gov/stations?state=OK&limit=500
